src/test/java/io/camunda/testing/utils/ProcessEventInspectionsTest.java

```diff
@@ -0,0 +1,92 @@
+package io.camunda.testing.utils;
+
+import static io.camunda.testing.util.Utilities.deployProcess;
+import static io.camunda.testing.util.Utilities.increaseTime;
+import static io.camunda.testing.utils.InspectionUtility.findProcessEvents;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.testing.extensions.ZeebeAssertions;
+import io.camunda.testing.util.Utilities.ProcessPackTimerStartEvent;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.DeploymentEvent;
+import java.time.Duration;
+import org.camunda.community.eze.RecordStreamSource;
+import org.camunda.community.eze.ZeebeEngine;
+import org.camunda.community.eze.ZeebeEngineClock;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@ZeebeAssertions
+class ProcessEventInspectionsTest {
+
+  private static final String WRONG_TIMER_ID = "wrongtimer";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+  private ZeebeEngineClock clock;
+
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    public void testAssertThatFirstProcessInstance() throws InterruptedException {
+      // given
+      final DeploymentEvent deploymentEvent =
+          deployProcess(client, ProcessPackTimerStartEvent.RESOURCE_NAME);
+
+      // when
+      increaseTime(clock, Duration.ofDays(1));
+
+      // then
+      findProcessEvents()
```

Comment:
:x: I am not a fan of this API. I would rather like to see something like:

```java
InstanceEvent processInstance = findProcessEvents()
          .triggeredByTimer(ProcessPackTimerStartEvent.TIMER_ID)
          .withProcessDefinitionKey(deploymentEvent.getProcesses().get(0).getProcessDefinitionKey())
          .firstProcessInstance();
          
assertThat(processInstance).isCompleted();
```

Reply:
So then the separation of concerns would be:
* Inspection utility is used to find an entity, that cannot be accessed in a different way
* BPMN assert then does assertions on that entity

Btw, I don't think the result of `firstProcessInstance()` needs to be an `InstanceEvent` if that is inconvenient. It could also return an `InspectedProcessInstance` and then have a different entry point in BPMNAssert